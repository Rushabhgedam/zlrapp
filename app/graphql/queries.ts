import { gql } from "@apollo/client";



export const GET_CONTININENTS = gql`query Continents {
    continents {
      name
      code
    }
  }`


export const GET_COUNTRIES_OF_CONTINENTS = gql`query Query($code: ID!) {
    continent(code: $code) {
      countries {
        name
        emoji
        continent {
          name
        }
      }
    }
  }
`