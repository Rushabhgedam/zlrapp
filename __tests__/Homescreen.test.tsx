

import { fireEvent, render, screen, waitFor } from "@testing-library/react-native";
import { GET_COUNTRIES_OF_CONTINENTS } from "../app/graphql/queries";

import { ApolloClient, ApolloProvider, InMemoryCache, useQuery, } from '@apollo/client';
import HomeScreen from "../app/homescreen";
import { MockedProvider } from '@apollo/client/testing';

const client = new ApolloClient({
  cache: new InMemoryCache(),
  uri: "https://countries.trevorblades.com/graphql"
})

const navigationMock = {
  navigate: jest.fn(),
};







jest.mock('react-native/Libraries/Utilities/Dimensions', () => ({
  get: jest.fn(() => ({ width: 360, height: 640 })),
}));


const continentMock = {
  code: 'AF',
  name: 'Africa',
};

const countriesMock = {
  continent: {
    countries: [
      {
        name: 'Uganda',
        emoji: '🌍',
        continent: {
          name: 'Africa',
        },
      },

    ],
  },
};

const mockData = {
  continent: {
    countries: [
      { name: 'MockCountry1', emoji: '🌍', continent: { name: 'MockContinent1' } },
      { name: 'MockCountry2', emoji: '🌎', continent: { name: 'MockContinent2' } },
    ],
  },
};


describe('HomeScreen', () => {
  it('Initial renders homescreen', async () => {
    const mock1 = [
      {
        request: {
          query: GET_COUNTRIES_OF_CONTINENTS,
          variables: { code: "AF" },
        },
        result: { data: mockData },
      }
    ];
    render(
      <MockedProvider mocks={mock1} addTypename={false}>
        <HomeScreen navigation={navigationMock} />
      </MockedProvider>
    );
    expect(screen).toMatchSnapshot()
  });


  test("Checks the component tree rendered correctly", async () => {
    const mock2 = [
      {
        request: {
          query: GET_COUNTRIES_OF_CONTINENTS,
          variables: { code: "AF" },
        },
        result: { data: mockData },
      }
    ];
    const { getByText, getByPlaceholderText, getByTestId } = render(
      <MockedProvider mocks={mock2} addTypename={false}>
        <HomeScreen navigation={navigationMock} />
      </MockedProvider>
    );
    expect(getByText('Select Continent')).toBeTruthy();
    fireEvent.press(getByTestId('continent0'));
    await waitFor(() => {
      expect(getByText('Available Countries in Africa are')).toBeTruthy();
    });
    const searchBar = getByPlaceholderText('Search Country here...')
    fireEvent.changeText(searchBar, "AF")
    expect(searchBar).toBeTruthy();
  })
});