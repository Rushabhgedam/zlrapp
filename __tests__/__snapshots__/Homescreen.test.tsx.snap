// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen Initial renders homescreen 1`] = `
<View
  style={
    {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "marginTop": 10,
      }
    }
  />
  <TextInput
    onChangeText={[Function]}
    placeholder="Please select continent"
    style={
      {
        "alignSelf": "center",
        "borderRadius": 8,
        "borderWidth": 0.5,
        "padding": 10,
        "width": "95%",
      }
    }
  />
  <View
    style={
      {
        "marginTop": 10,
      }
    }
  />
  <View>
    <RCTScrollView
      ItemSeparatorComponent={[Function]}
      data={
        [
          {
            "code": "AF",
            "name": "Africa",
          },
          {
            "code": "AN",
            "name": "Antarctica",
          },
          {
            "code": "AS",
            "name": "Asia",
          },
          {
            "code": "EU",
            "name": "Europe",
          },
          {
            "code": "NA",
            "name": "North America",
          },
          {
            "code": "OC",
            "name": "Oceania",
          },
          {
            "code": "SA",
            "name": "South America",
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={[]}
      viewabilityConfigCallbackPairs={[]}
    >
      <View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "white",
                "borderRadius": 6,
                "flexDirection": "row",
                "marginHorizontal": 6,
                "opacity": 1,
                "paddingHorizontal": 6,
              }
            }
            testID="continent0"
          >
            <Image
              source={
                {
                  "testUri": "../../../app/assets/unselected.png",
                }
              }
              style={
                {
                  "height": 20,
                  "resizeMode": "contain",
                  "tintColor": "skyblue",
                  "width": 20,
                }
              }
            />
            <Text
              style={
                {
                  "color": "black",
                }
              }
            >
              Africa
            </Text>
          </View>
          <View
            style={
              {
                "paddingVertical": 2,
              }
            }
          />
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "white",
                "borderRadius": 6,
                "flexDirection": "row",
                "marginHorizontal": 6,
                "opacity": 1,
                "paddingHorizontal": 6,
              }
            }
            testID="continent1"
          >
            <Image
              source={
                {
                  "testUri": "../../../app/assets/unselected.png",
                }
              }
              style={
                {
                  "height": 20,
                  "resizeMode": "contain",
                  "tintColor": "skyblue",
                  "width": 20,
                }
              }
            />
            <Text
              style={
                {
                  "color": "black",
                }
              }
            >
              Antarctica
            </Text>
          </View>
          <View
            style={
              {
                "paddingVertical": 2,
              }
            }
          />
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "white",
                "borderRadius": 6,
                "flexDirection": "row",
                "marginHorizontal": 6,
                "opacity": 1,
                "paddingHorizontal": 6,
              }
            }
            testID="continent2"
          >
            <Image
              source={
                {
                  "testUri": "../../../app/assets/unselected.png",
                }
              }
              style={
                {
                  "height": 20,
                  "resizeMode": "contain",
                  "tintColor": "skyblue",
                  "width": 20,
                }
              }
            />
            <Text
              style={
                {
                  "color": "black",
                }
              }
            >
              Asia
            </Text>
          </View>
          <View
            style={
              {
                "paddingVertical": 2,
              }
            }
          />
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "white",
                "borderRadius": 6,
                "flexDirection": "row",
                "marginHorizontal": 6,
                "opacity": 1,
                "paddingHorizontal": 6,
              }
            }
            testID="continent3"
          >
            <Image
              source={
                {
                  "testUri": "../../../app/assets/unselected.png",
                }
              }
              style={
                {
                  "height": 20,
                  "resizeMode": "contain",
                  "tintColor": "skyblue",
                  "width": 20,
                }
              }
            />
            <Text
              style={
                {
                  "color": "black",
                }
              }
            >
              Europe
            </Text>
          </View>
          <View
            style={
              {
                "paddingVertical": 2,
              }
            }
          />
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "white",
                "borderRadius": 6,
                "flexDirection": "row",
                "marginHorizontal": 6,
                "opacity": 1,
                "paddingHorizontal": 6,
              }
            }
            testID="continent4"
          >
            <Image
              source={
                {
                  "testUri": "../../../app/assets/unselected.png",
                }
              }
              style={
                {
                  "height": 20,
                  "resizeMode": "contain",
                  "tintColor": "skyblue",
                  "width": 20,
                }
              }
            />
            <Text
              style={
                {
                  "color": "black",
                }
              }
            >
              North America
            </Text>
          </View>
          <View
            style={
              {
                "paddingVertical": 2,
              }
            }
          />
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "white",
                "borderRadius": 6,
                "flexDirection": "row",
                "marginHorizontal": 6,
                "opacity": 1,
                "paddingHorizontal": 6,
              }
            }
            testID="continent5"
          >
            <Image
              source={
                {
                  "testUri": "../../../app/assets/unselected.png",
                }
              }
              style={
                {
                  "height": 20,
                  "resizeMode": "contain",
                  "tintColor": "skyblue",
                  "width": 20,
                }
              }
            />
            <Text
              style={
                {
                  "color": "black",
                }
              }
            >
              Oceania
            </Text>
          </View>
          <View
            style={
              {
                "paddingVertical": 2,
              }
            }
          />
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "white",
                "borderRadius": 6,
                "flexDirection": "row",
                "marginHorizontal": 6,
                "opacity": 1,
                "paddingHorizontal": 6,
              }
            }
            testID="continent6"
          >
            <Image
              source={
                {
                  "testUri": "../../../app/assets/unselected.png",
                }
              }
              style={
                {
                  "height": 20,
                  "resizeMode": "contain",
                  "tintColor": "skyblue",
                  "width": 20,
                }
              }
            />
            <Text
              style={
                {
                  "color": "black",
                }
              }
            >
              South America
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
  <View
    style={
      {
        "marginTop": 10,
      }
    }
  />
  <View>
    <RCTScrollView
      ListEmptyComponent={[Function]}
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      refreshControl={
        <RefreshControlMock
          onRefresh={[Function]}
          refreshing={true}
        />
      }
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={[]}
      style={
        {
          "marginBottom": 224,
          "marginHorizontal": 6,
        }
      }
      viewabilityConfigCallbackPairs={[]}
    >
      <RCTRefreshControl />
      <View>
        <View>
          <Text>
            Select Continent
          </Text>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
